corgi/collectors/brew.py

```diff
@@ -666,21 +671,45 @@ def get_component_data(self, build_id: int) -> dict:
 
         # Determine build type
         build_type_info = self.koji_session.getBuildType(build)
+        build_type = next(
+            type_ for type_ in build_type_info.keys() if type_ in self.SUPPORTED_BUILD_TYPES
+        )
         if not any(type_ in self.SUPPORTED_BUILD_TYPES for type_ in build_type_info.keys()):
             raise BrewBuildTypeNotSupported(
                 f"Build {build_id} type is not supported: {build_type_info}"
             )
-        build_type = next(
-            type_ for type_ in build_type_info.keys() if type_ in self.SUPPORTED_BUILD_TYPES
-        )
+        # TODO: refactor Brew.CONTAINER_BUILD_TYPE to be a generic IMAGE_TYPE with image types
+        #  identified in a separate attribute on the build itself.
+        if build_type == self.CONTAINER_BUILD_TYPE:
+            # If this is an "image" type build, it may be building a container image, ISO image,
+            # or other types of images. Check the content generator name to determine where this
+            # image was built, which indicates what type of image it is.
+            if build["cg_name"] != "atomic-reactor":
+                # Container images are built in OSBS, which uses atomic-reactor to build them.
+                raise BrewBuildTypeNotSupported(
+                    f"Image build {build_id} is not supported: "
+                    f"{build['cg_name']} content generator used"
+                )
         build["type"] = build_type
 
         # Determine build source
         if not build.get("source"):
-            # Sometimes there is no source URL on the build but it can be found in the task
+            # Sometimes there is no source URL on the build, but it can be found in the task
             # request info instead.
             logger.info("Fetching source from task info for %s", build_id)
-            build["source"] = self.get_source_of_build(build)
+            try:
+                build["source"] = self.get_source_of_build(build)
+            except BrewBuildSourceNotFound as exc:
+                # Some older builds do not specify source URLs; the below date was chosen based
+                # on some initial analysis of source-less builds in Brew.
+                if datetime.fromtimestamp(build["completion_ts"]) < datetime(2015, 1, 1):
+                    logger.error(
+                        f"Build {build_id} has no associated source URL but is too old "
+                        f"to process; returning an empty component."
+                    )
+                    return {}
```

Comment:
Just saw this change (was looking at old diff before). Are we really unable to do anything without a source URL? I guess we can't do SCA without something to download, but we still can get tags and errata_tags. Does build_data below depend on having a source_url to analyze?

If not, could we still get all other data besides the source URL, then return the built component dict's data (just missing a source URL) instead of an empty dict? It seems more useful to try and return something. But I'm not really sure how the source URL is used, how we want to indicate "this build has no source", and what the dependencies are. So it's fine to leave it like this if you think that's better.